crates/cargo-wdk/src/actions/new/new_driver.rs

```diff
@@ -0,0 +1,276 @@
+    /// * `driver_type` - The type of the driver.
+    /// * `cwd` - The current working directory.
+    /// * `command_exec` - The command execution provider.
+    /// * `fs_provider` - The file system provider.
+    ///
+    /// # Returns
+    ///
+    /// * `Self` - A new instance of `NewDriver`.
+    pub fn new(
+        driver_project_name: &'a str,
+        driver_type: DriverType,
+        cwd: &'a Path,
+        command_exec: &'a dyn RunCommand,
+        fs_provider: &'a dyn FSProvider,
+    ) -> Self {
+        let cwd = cwd.join(driver_project_name);
+        let driver_project_name = driver_project_name.replace('-', "_");
+        Self {
+            driver_project_name,
+            driver_type,
+            cwd,
+            command_exec,
+            fs_provider,
+        }
+    }
+
+    /// Runs the action to create a new driver project.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), NewDriverError>` - A result indicating success or failure
+    ///   of the action.
+    ///
+    /// # Errors
+    ///
+    /// * `NewDriverError::CargoNewCommand` - If there is an error running the
+    ///   `cargo new` command.
+    /// * `NewDriverError::TemplateNotFound` - If a required template file is
+    ///   not found.
+    /// * `NewDriverError::FileSystem` - If there is an error with file system
+    ///   operations.
+    pub fn run(&self) -> Result<(), NewDriverError> {
+        debug!("Creating new project");
+        self.run_cargo_new()?;
+        self.copy_lib_rs_template()?;
+        self.update_cargo_toml()?;
+        self.create_inx_file()?;
+        self.copy_build_rs_template()?;
+        if matches!(self.driver_type, DriverType::Kmdf | DriverType::Wdm) {
+            self.copy_cargo_config()?;
+        }
+        info!(
+            "New Driver Project {} created at {}",
+            self.driver_project_name,
+            self.cwd.display()
+        );
+        Ok(())
+    }
+
+    /// Runs the `cargo new` command to create a new Rust library project.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<(), NewDriverError>` - A result indicating success or failure
+    ///   of the command.
+    ///
+    /// # Errors
+    ///
+    /// * `NewDriverError::CargoNewCommand` - If there is an error running the
+    ///   `cargo new` command.
+    fn run_cargo_new(&self) -> Result<(), NewDriverError> {
+        debug!(
+            "Running cargo new for project: {}",
+            self.driver_project_name
+        );
+        let args = ["new", "--lib", &self.driver_project_name, "--vcs", "none"];
+        if let Err(e) = self.command_exec.run("cargo", &args, None) {
+            return Err(NewDriverError::CargoNewCommand(e));
+        }
+        debug!(
+            "Successfully ran cargo new for project: {}",
+            self.driver_project_name
+        );
+        Ok(())
+    }
+
+    /// Copies the `lib.rs` template for the specified driver type.
```

Comment:
Copies where? It seems AI is simply looking at the name of the function and expanding that into a sentence without adding any meaning. This seems true of many comments here.

Reply:
Yes true, we will revisit these auto generated comments